File: lab4/backend/products/soa-products-rest-adapter/src/main/kotlin/com/soa/products/rest/adapter/wsdl/FilterOperatorTo.java

package com.soa.products.rest.adapter.wsdl;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlType;


/**
 * 
 * 
 * <p>Java class for filterOperatorTo</p>.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.</p>
 * <pre>{@code
 * <simpleType name="filterOperatorTo">
 *   <restriction base="{http://www.w3.org/2001/XMLSchema}string">
 *     <enumeration value="EQUAL"/>
 *     <enumeration value="LESS"/>
 *     <enumeration value="GREATER"/>
 *     <enumeration value="LESS_OR_EQUAL"/>
 *     <enumeration value="GREATER_OR_EQUAL"/>
 *     <enumeration value="SAME"/>
 *   </restriction>
 * </simpleType>
 * }</pre>
 * 
 */
@XmlType(name = "filterOperatorTo")
@XmlEnum
public enum FilterOperatorTo {

    EQUAL,
    LESS,
    GREATER,
    LESS_OR_EQUAL,
    GREATER_OR_EQUAL,
    SAME;

    public String value() {
        return name();
    }

    public static FilterOperatorTo fromValue(String v) {
        return valueOf(v);
    }

}

File: lab4/backend/products/soa-products/src/main/kotlin/com/soa/products/dto/FilterOperatorTo.kt
package com.soa.products.dto

enum class FilterOperatorTo(val value: String) {

    EQUAL("EQUAL"),
    LESS("LESS"),
    GREATER("GREATER"),
    LESS_OR_EQUAL("LESS_OR_EQUAL"),
    GREATER_OR_EQUAL("GREATER_OR_EQUAL"),
    SAME("SAME");

    companion object {
        fun forValue(value: String): FilterOperatorTo {
            return values().firstOrNull { it.value == value }
                ?: throw IllegalArgumentException("No enum constant with value $value")
        }
    }
}
